src/resources/touch_esp32.cc

```diff
@@ -0,0 +1,262 @@
+// Copyright (C) 2021 Toitware ApS.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; version
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
+#include "../top.h"
+
+#ifdef TOIT_FREERTOS
+
+#include <driver/gpio.h>
+#include <driver/touch_sensor.h>
+
+#include "touch_esp32.h"
+
+#include "../entropy_mixer.h"
+#include "../objects_inline.h"
+#include "../primitive.h"
+#include "../process.h"
+#include "../resource.h"
+#include "../resource_pool.h"
+#include "../vm.h"
+
+#include "../event_sources/gpio_esp32.h"
+#include "../event_sources/system_esp32.h"
+
+namespace toit {
+
+#ifdef CONFIG_IDF_TARGET_ESP32
+
+static touch_pad_t get_touch_pad(int pin) {
+  switch (pin) {
+    case 0: return TOUCH_PAD_NUM1;
+    case 2: return TOUCH_PAD_NUM2;
+    case 4: return TOUCH_PAD_NUM0;
+    case 12: return TOUCH_PAD_NUM5;
+    case 13: return TOUCH_PAD_NUM4;
+    case 14: return TOUCH_PAD_NUM6;
+    case 15: return TOUCH_PAD_NUM3;
+    case 27: return TOUCH_PAD_NUM7;
+    case 32: return TOUCH_PAD_NUM9;
+    case 33: return TOUCH_PAD_NUM8;
+    default: return touch_pad_t(-1);
+  }
+}
+
+int touch_pad_to_pin_num(touch_pad_t pad) {
+  switch (pad) {
+    case TOUCH_PAD_NUM1: return 0;
+    case TOUCH_PAD_NUM2: return 2;
+    case TOUCH_PAD_NUM0: return 4;
+    case TOUCH_PAD_NUM5: return 12;
+    case TOUCH_PAD_NUM4: return 13;
+    case TOUCH_PAD_NUM6: return 14;
+    case TOUCH_PAD_NUM3: return 15;
+    case TOUCH_PAD_NUM7: return 27;
+    case TOUCH_PAD_NUM9: return 32;
+    case TOUCH_PAD_NUM8: return 33;
+    default: return -1;
+  }
+}
+
+#elif CONFIG_IDF_TARGET_ESP32S2
+
+static touch_pad_t get_touch_pad(int pin) {
+  switch (pin) {
+    case 1: return TOUCH_PAD_NUM1;
```

Comment:
Want to align the returns like in touch_pad_to_pin_num?

Reply:
done.